containers/java-postgres/test-project/src/test/java/com/mycompany/app/AppTest.java

```diff
@@ -0,0 +1,125 @@
+ *-------------------------------------------------------------------------------------------------------------*/
+
+package com.mycompany.app;
+
+import org.junit.Test;
+
+import junit.framework.AssertionFailedError;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+public class AppTest
+{
+
+
+    public static boolean pingAddress(String ipAddress) throws UnknownHostException, IOException
+    {
+        InetAddress postgresAddress = InetAddress.getByName(ipAddress);
+        System.out.println("Sending Ping Request to " + ipAddress);
+        if (postgresAddress.isReachable(5000)){
+            System.out.println("Successfully Reached: " + ipAddress);
+            return true;
+        }
+        System.out.println("Could not reach or connect to: " + ipAddress);
+        return false;
+    }
+
+    public static boolean dbLogin(String ipAddress, String username, String password){
+        Connection c = null;
+        try {
+            Class.forName("org.postgresql.Driver");
+            c = DriverManager
+               .getConnection("jdbc:postgresql://postgresdb/postgres",
+               "postgres", "postgres");
+            return true;
+         } catch (Exception e) {
+            e.printStackTrace();
+            System.err.println(e.getClass().getName()+": "+e.getMessage());
+            return false;
+         }
+    }
+
+    public static boolean listDatabases(String ipAddress, String username, String password){
+        Connection c = null;
+        Statement stmt = null;
+
+        try {
+            Class.forName("org.postgresql.Driver");
+            c = DriverManager
+               .getConnection("jdbc:postgresql://postgresdb/postgres",
+               "postgres", "postgres");
+            c.setAutoCommit(false);
+            stmt = c.createStatement();
+            ResultSet rs = stmt.executeQuery( "select * from pg_database;" );
+            System.out.println("List of databases in this cluster.");
+            while (rs.next()){
+                String databaseName = rs.getString("datname");
+                System.out.printf("Database Name = %s ", databaseName);
+                System.out.println();
+            }
+            return true;
+         } catch (Exception e) {
+            e.printStackTrace();
+            System.err.println(e.getClass().getName()+": "+e.getMessage());
+            return false;
+         }
+    }
+
+
+    App newApp = new App();
+    public AppTest() {
+    }
+
+    @Test
+    public void testApp()
+    {
+        assertTrue( true );
+    }
+
+    @Test
+    public void testMore()
+    {
+        assertTrue( true );
+    }
+
+    @Test
+    public void testIP()
+    {
+        try {
+            assertTrue( pingAddress(System.getenv("POSTGRES_IP")) );
```

Comment:
Don't see the benefit of invoking static methods into the test method. It would be simpler to just have here the full implementation.

Reply:
I could do that.

Reply:
hi @juzuluag , part of the reason the 3 methods are static was we wanted them to live in App.java, and be used both by "F5 Debug" of the main app, and be reused the Test-cases -- I ran into an issue getting `./test.sh` to work correctly though, which some research makes me think was apparently due the result of how maven builds the jar..., since the error only would occur when running `test.sh`, or `java -jar target/my-app-1.0-SNAPSHOT.jar` manually, but not when using right-click "Run Java" or "F5 Debug" ... :

🧪 Testing test-project
Hello Remote World!
The XML namespace prefix is: xml
java.lang.ClassNotFoundException: org.postgresql.Driver
at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
at java.base/java.lang.Class.forName0(Native Method)
at java.base/java.lang.Class.forName(Class.java:315)
at com.mycompany.app.App.dbLogin(App.java:55)
at com.mycompany.app.App.main(App.java:30)